pkg/tsdb/interval/interval.go

```diff
@@ -69,10 +69,15 @@ func (ic *intervalCalculator) Calculate(timerange plugins.DataTimeRange, minInte
 func GetIntervalFrom(dsInfo *models.DataSource, queryModel *simplejson.Json, defaultInterval time.Duration) (time.Duration, error) {
 	interval := queryModel.Get("interval").MustString("")
 
-	if interval == "" && dsInfo != nil && dsInfo.JsonData != nil {
-		dsInterval := dsInfo.JsonData.Get("timeInterval").MustString("")
-		if dsInterval != "" {
-			interval = dsInterval
+	if interval == "" {
+		intervalMs := queryModel.Get("intervalMs").MustInt(0)
+		if intervalMs != 0 {
+			return time.Duration(intervalMs * 1000 * 1000), nil
+		} else if dsInfo != nil && dsInfo.JsonData != nil {
+			dsInterval := dsInfo.JsonData.Get("timeInterval").MustString("")
+			if dsInterval != "" {
+				interval = dsInterval
+			}
```

Comment:
* Nit: Initialisms should be written in [consistent case](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) in Go
* It's much more robust and readable to use `time` constants instead of their numeric equivalents:

```suggestion
		intervalMS := queryModel.Get("intervalMs").MustInt(0)
		if intervalMS != 0 {
			return time.Duration(intervalMS * time.Millisecond), nil
		}

		if dsInfo != nil && dsInfo.JsonData != nil {
			dsInterval := dsInfo.JsonData.Get("timeInterval").MustString("")
			if dsInterval != "" {
				interval = dsInterval
			}
```

Reply:
Thanks for the review - still getting used to Go idioms. I've pushed your suggestions.